website/src/parsers/js/hermes.js

```diff
@@ -0,0 +1,44 @@
+import defaultParserInterface from './utils/defaultESTreeParserInterface';
+import HermesWorkerClient from './hermes/HermesWorkerClient';
+
+const ID = 'hermes';
+export const defaultOptions = {
+  jsx: true,
+};
+export const parserSettingsConfiguration = {
+  fields: [
+    'jsx',
+  ],
+};
+
+export default {
+  ...defaultParserInterface,
+
+  id: ID,
+  displayName: ID,
+  version: 'N/A',
+  homepage: 'https://hermesengine.dev/',
+  locationProps: new Set(['loc', 'range']),
+
+  loadParser(callback) {
+    callback(new HermesWorkerClient());
+  },
+
+  async parse(hermes, code, options) {
+    const hermesOutput = await hermes.run(['-dump-ast', '--pretty-json', '-dump-source-location', options.jsx ? '-parse-jsx' : ''].filter(Boolean), code);
```

Comment:
Does Hermes not provide a library-based API, only a binary?

Reply:
I believe we're in the process of figuring out APIs for various similar integrations, not sure if there's anything very concrete yet that would fit here. (cc @tmikov)

Reply:
I think ideally we should do something like https://github.com/facebook/hermes/blob/master/tools/emhermesc/HermesCompiler.js , but linking only the parser instead of the whole compiler and VM.

Even more ideally :-), we can construct the JS AST by calling JS callbacks from WASM instead of going through JSON, though wonder how much faster it would be. It shouldn't matter for this particular use case.

We really should settle these interfaces soon, before we publish an NPM and unblock this PR (in some form).